src/ray/core_worker/core_worker.cc

```diff
@@ -677,6 +685,63 @@ void CoreWorker::InternalHeartbeat(const boost::system::error_code &error) {
   internal_timer_.async_wait(boost::bind(&CoreWorker::InternalHeartbeat, this, _1));
 }
 
+void CoreWorker::LocationResolveHeartBeat(const boost::system::error_code &error) {
+  if (error == boost::asio::error::operation_aborted) {
+    return;
+  }
+  // Note: This will poll the GCS service every certain period.
+  // This exists to resolve scenarios such as
+  // https://github.com/ray-project/ray/pull/8679/files
+  ResolveActorsLocationNotPersistedToGCS();
+  internal_timer_.expires_at(
+      internal_timer_.expiry() +
+      boost::asio::chrono::milliseconds(kLocationResolveHeartbeatMillis));
+  internal_timer_.async_wait(
+      boost::bind(&CoreWorker::LocationResolveHeartBeat, this, _1));
+}
+
+void CoreWorker::ResolveActorsLocationNotPersistedToGCS() {
+  absl::MutexLock lock(&actor_location_resolve_waiters_mutex_);
+  for (auto it = actor_location_resolve_waiters_.begin();
+       it != actor_location_resolve_waiters_.end();) {
+    auto current = it++;
+    const auto &actor_id = *current;
+    ActorHandle *actor_handle = nullptr;
+    RAY_CHECK_OK(GetActorHandle(actor_id, &actor_handle));
+
+    if (actor_handle->IsPersistedToGCS()) {
+      // if an actor is already persisted to GCS, GCS will be responsible for reporting
+      // state of actors, so just remove it from the waiter.
+      actor_location_resolve_waiters_.erase(actor_id);
+    } else {
+      // If it is still not persisted to GCS, check worker/node is still alive.
+      // Disconnect actors if the worker or node is dead.
+      // This is needed because until the owner's local depedencies are resolved,
+      // actors won't be persisted to GCS, and there's no way we can know the states of
+      // them. Question? Is is possible worker is restarted after reporting failures?
```

Comment:
It can get restarted, but it'll get restarted with a different worker ID and attempt to create the actor again.